rotkehlchen/tests/unit/decoders/test_thegraph.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.evm_event import EvmEvent
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.modules.thegraph.constants import CPT_THEGRAPH
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_ETH, A_GRT
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+ADDY_USER = string_to_evm_address('0xd200aeEC7Cd9dD27CAB5a85083953a734D4e84f0')
+ADDY_THEGRAPH = string_to_evm_address('0xF55041E37E12cD407ad00CE2910B8269B01263b9')
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY_USER]])
+def test_thegraph_delegate(database, ethereum_inquirer):
+    tx_hash = deserialize_evm_tx_hash('0x6ed3377db652151fb8e4794dd994a921a2d029ad317bd3f2a2916af239490fec')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1690731467000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.002150596408306665')),
+            location_label=ADDY_USER,
+            notes='Burned 0.002150596408306665 ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=357,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.STAKING,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_GRT,
+            balance=Balance(amount=FVal('998.98')),
+            location_label=ADDY_USER,
+            notes='Delegated 998.98 GRT to indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7',
+            counterparty=CPT_THEGRAPH,
+            address=ADDY_THEGRAPH,
+        ), EvmEvent(
```

Comment:
This approval event in between is annoying. Can you use `maybe_reshuffle_events` in the decoder to guarantee Delegate and burn come one after the other?